src/libclipper/test/json_util_test.cpp

```diff
@@ -0,0 +1,206 @@
+#include <gtest/gtest.h>
+
+#include <clipper/datatypes.hpp>
+#include <clipper/json_util.hpp>
+
+using namespace clipper_json;
+
+/* Test JSON serialization utilities */
+TEST(JsonUtilTests, TestCorrectJsonValues) {
+  std::string expected_json =
+      "{\"string_val\":\"test_string\",\"double_val\":0.3,\"int_val\":-100}";
+  std::string string_val = "test_string";
+  double double_val = 0.3;
+  int int_val = -100;
+
+  rapidjson::Document d;
+  d.SetObject();
+  add_string(string_val, "string_val", d);
+  add_double(double_val, "double_val", d);
+  add_int(int_val, "int_val", d);
+  std::string output_json = to_json_string(d);
+  EXPECT_EQ(output_json, expected_json); 
+}
+
+TEST(JsonUtilTests, TestCorrectJsonArrays) {
+  std::string expected_json =
+      "{\"string_array\":[\"test1\",\"word2\",\"phrase3\"],"
+       "\"double_array\":[1.4,2.23,3.243242,0.3223424],"
+       "\"int_array\":[1,2,3,4]}";
+  std::vector<std::string> string_array = {"test1", "word2", "phrase3"};
+  std::vector<double> double_array = {1.4,2.23,3.243242,0.3223424};
+  std::vector<int> int_array = {1, 2, 3, 4};
+
+  rapidjson::Document d;
+  d.SetObject();
+  add_string_array(string_array, "string_array", d);
+  add_double_array(double_array, "double_array", d);
+  add_int_array(int_array, "int_array", d);
+  std::string output_json = to_json_string(d);
+  EXPECT_EQ(output_json, expected_json); 
+}
+
+TEST(JsonUtilTests, TestCorrectJsonNestedObjects) {
```

Comment:
@Corey-Zumar In contrast to the example below, this is an example of how you'd go about creating nested JSON objects.